pkg/sql/alter_external_connection.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package sql
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/cockroachdb/cockroach/pkg/cloud/externalconn"
+	"github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgcode"
+	"github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgerror"
+	"github.com/cockroachdb/cockroach/pkg/sql/privilege"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/sql/sessiondata"
+	"github.com/cockroachdb/cockroach/pkg/sql/syntheticprivilege"
+	"github.com/cockroachdb/errors"
+)
+
+const alterExternalConnectionOp = "ALTER EXTERNAL CONNECTION"
+
+type alterExternalConnectionNode struct {
+	zeroInputPlanNode
+	n *tree.AlterExternalConnection
+}
+
+func (p *planner) AlterExternalConnection(
+	ctx context.Context, n *tree.AlterExternalConnection,
+) (planNode, error) {
+	return &alterExternalConnectionNode{n: n}, nil
+}
+
+func (alt *alterExternalConnectionNode) startExec(params runParams) error {
+	return params.p.alterExternalConnection(params, alt.n)
+}
+
+func (p *planner) alterExternalConnection(params runParams, n *tree.AlterExternalConnection) error {
+	txn := p.InternalSQLTxn()
+	if err := params.p.CheckPrivilege(params.ctx, syntheticprivilege.GlobalPrivilegeObject, privilege.EXTERNALCONNECTION); err != nil {
+		return pgerror.New(
+			pgcode.InsufficientPrivilege,
+			"only users with the EXTERNALCONNECTION system privilege are allowed to ALTER EXTERNAL CONNECTION",
+		)
+	}
+
+	exprEval := p.ExprEvaluator(alterExternalConnectionOp)
+	name, err := exprEval.String(params.ctx, n.ConnectionLabelSpec.Label)
+	if err != nil {
+		return err
+	}
+	endpoint, err := exprEval.String(params.ctx, n.As)
+	if err != nil {
+		return err
+	}
+
+	ec, err := externalconn.LoadExternalConnection(params.ctx, name, txn)
+	if err != nil {
+		if ok := strings.Contains(err.Error(), "does not exist"); ok && n.IfExists {
+			return nil
+		}
+		return err
+	}
+
+	ex := externalconn.NewMutableExternalConnection()
+	ex.SetConnectionName(ec.ConnectionName())
+	if err = logAndSanitizeExternalConnectionURI(params.ctx, endpoint); err != nil {
+		return errors.Wrap(err, "failed to log and santitize External Connection")
+	}
+
+	var skipCheckingExternalStorageConnection bool
+	var skipCheckingKMSConnection bool
+	if tk := params.ExecCfg().ExternalConnectionTestingKnobs; tk != nil {
```

Comment:
since we arent using these testing knobs in this code, i don't think we need to have them here